### ISSUE TITLE
Devices

### ISSUE THREAD

**Issue by mabunixda** [paultyng/terraform-provider-unifi#15]
Hi,

i saw already the json definitions in your sdk. so is it just a matter of implementing the go code in the terraform provider, to get also devices configurable? 
Even I am missing the accesspoint stuff - usg and usw seems to be available.

Thanks, Martin

**Comment by paultyng** [comment]
I believe so, the "settings" stuff is hard to follow a little, and I haven't played around with it much. The way I mostly do this is connect to the my controller via IP address and look at the XHR requests in the browser to see what is being sent and received then try to emulate that over the SDK. It may require changes to the SDK as well, specifically I run in to marshalling problems as Unifi's marshalling for things like ints is sometimes weird (ie, they will return an empty string or quoted number occasionally).

**Comment by mabunixda** [comment]
Can you create a short update on the SDK how to extend it? Or did I miss it in the readme?

**Comment by dekarl** [comment]
(y) This would be a great feature. I have just reset my controller and stumbled upon the click-fest that is configuring all radios, again...

I'd love to apply a common set of radio settings to my APs. Basically iterating over the "All APs" ap_group and configuring them according to [optimising-unifi-performance](https://github.com/adrianmace/blog/blob/master/optimising-unifi-performance.md)
(leaving the radio channel alone for now)

**Comment by paultyng** [comment]
I'm definitely open to adding this, I haven't found an easy way to simulate adoption for automated testing though, so makes it a bit harder to work with if this would require adoption to use those API endpoints in the controller (similar to the RADIUS testing in #30). I don't have a test setup at home, just my actual wifi, so testing on that upsets the family 😊.

**Comment by paultyng** [comment]
Switches and port configuration is in. Its kind of weird as you can't create or destroy the resource, so you have to `terraform import` it but I'm using this to manage port profiles on my switch ports now successfully. I haven't figured out testing on it quite yet, still investigating that.

**Comment by joshuaspence** [comment]
I don't have time to debug it at the moment but this isn't working for me. It fails on import:

```
Error: unable to decode body: GET s/default/stat/device unable to unmarshal alias: unable to unmarshal alias: json: cannot unmarshal number into Go struct field .channel of type string
```

**Comment by paultyng** [comment]
Argh, this is the problem with limited testing of these adopted things, I know what this is though, let me get a hotfix up.

**Comment by paultyng** [comment]
If you have the capability to run a custom build you can try #120, I fixed the marshalling I hope to handle both numbers and strings for the channel fields (they can be JSON numbers or the string `"auto"`).

**Comment by paultyng** [comment]
Sorry, build error, let me bump it again real quick.

**Comment by joshuaspence** [comment]
I get a different error now...

```
Error: unable to decode body: GET s/default/stat/device unable to unmarshal alias: unable to unmarshal alias: json: cannot unmarshal number into Go struct field .tx_power of type string
```

**Comment by paultyng** [comment]
Argh, this is going to be whack-a-mole I guess.  `TxPower` is another one that allows for a number or `"auto"`.

**Comment by paultyng** [comment]
It looks like `TxPower` is the only other occurrence, let me bump the SDK again.

**Comment by paultyng** [comment]
New version pushed.

**Comment by joshuaspence** [comment]
Works for me now, thanks

**Comment by paultyng** [comment]
Great, will ship a hotfix release of the provider. Thanks for testing!

**Comment by dunderrrrrr** [comment]
Seems like this is still an issue? I'm having trouble importing a device.

```
│ Error: unable to decode body: GET s/default/stat/device unable to unmarshal alias: json: cannot unmarshal string into Go struct field .x of type float64
```

What's the `.x` all about? I'm on a controller with version `7.3.83`.
